string/words.js

```diff
@@ -0,0 +1,19 @@
+/**
+Extracts words from str.
+@module string/words
+@param {string} str The string from which to extract the words.
+@param {pattern} [pattern=] The pattern to match the words.
```

Comment:
I'm not too sure how to communicate that the parameter is optional. Please advise.